### PR TITLE
Fix bug in constructing Graph from CircuitSeq

### PR DESCRIPTION
**Pull request by zikun-li**

### CODE REVIEW

**Comment by zikun-li** [comment]
This pull request tries to fix the bug in #122.

File: src/quartz/tasograph/tasograph.cpp [hunk 1]
```diff
@@ -112,46 +102,69 @@ Graph::Graph(Context *ctx, const CircuitSeq *seq)
...
+          }
+        }
+        assert(found);
+        assert(edge_2_op.find(output_edge) != edge_2_op.end());
+        auto dstOp = edge_2_op[output_edge];
 
-      add_edge(srcOp, dstOp, srcIdx, dstIdx);
+        Op srcOp = Op(context->next_global_unique_id(),
+                      context->get_gate(GateType::input_param));
+        constant_param_values[srcOp] = context->input_parameters[node->index];
```
Comment (xumingkuan) [inline src/quartz/tasograph/tasograph.cpp:165]: Let's assert that the context has this parameter before this line:
```suggestion
        assert(node->index < (int)context->input_parameters.size());
        constant_param_values[srcOp] = context->input_parameters[node->index];
```

**Review by xumingkuan — COMMENTED** [review]
Thanks!

**Review by xumingkuan — APPROVED** [review]

**Comment by typerSniper** [comment]
does this fix the issue? If so, let's close it!

**Comment by zikun-li** [comment]
I think this could fix #122.

**Pull request merged**